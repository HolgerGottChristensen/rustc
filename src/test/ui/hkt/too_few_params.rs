// HKT test for too few parameters
// check-fail

fn test<I<%J, %K>>(input: I<u32>) {} //~ ERROR this hkt parameter takes 2 generic arguments but 1 generic argument was supplied [E0107]

fn main() {}
