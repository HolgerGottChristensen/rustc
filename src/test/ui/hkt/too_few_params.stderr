error[E0107]: this hkt parameter takes 2 generic arguments but 1 generic argument was supplied
  --> $DIR/too_few_params.rs:4:27
   |
LL | fn test<I<%J, %K>>(input: I<u32>) {}
   |                           ^ --- supplied 1 generic argument
   |                           |
   |                           expected 2 generic arguments
   |
note: hkt parameter defined here, with 2 generic parameters: `J`, `K`
  --> $DIR/too_few_params.rs:4:9
   |
LL | fn test<I<%J, %K>>(input: I<u32>) {}
   |         ^  -   -
help: add missing generic argument
   |
LL | fn test<I<%J, %K>>(input: I<u32, K>) {}
   |                                +++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0107`.
